klayout_package/python/kqcircuits/simulations/simulation.py

```diff
@@ -1455,3 +1457,73 @@ def visualise_region(
+
+    for port in self.ports:
+        # Try to find a matching entry in port_json
+        port_data_list = [p for p in port_json if p.get("number") == port.number]
+        if not port_data_list:
+            logging.warning(f"Port {port.number} not found in get_port_data() output, skipping visualisation.")
+            continue
+        port_data = port_data_list[0]
+
+        # Label based on port type
+        if isinstance(port, EdgePort):
+            label = f"edge_port_{port.number}"
+        elif isinstance(port, InternalPort):
+            label = f"internal_port_{port.number}"
+        else:
+            logging.warning(f"Unsupported port type {type(port).__name__} for port {port.number}")
+            continue
+
+        # Construct 2D polygon from port_data["polygon"] (ignoring z coordinate)
+        if "polygon" in port_data and port_data["polygon"]:
+            points_2d = [pya.DPoint(p[0], p[1]) for p in port_data["polygon"]]
+
+            if isinstance(port, EdgePort) and edge_port_thickness > 0:
+                    direction = None
+                    if port.signal_location.x == self.box.p1.x:
+                        # Port on left border of simulation box
+                        direction = pya.DPoint(-edge_port_thickness, 0)
+                    elif port.signal_location.x == self.box.p2.x:
+                        # on right border
+                        direction = pya.DPoint(edge_port_thickness, 0)
+                    elif port.signal_location.y == self.box.p1.y:
+                        # on bottom border
+                        direction = pya.DPoint(0, -edge_port_thickness)
+                    elif port.signal_location.y == self.box.p2.y:
+                        # on top border
+                        direction = pya.DPoint(0, edge_port_thickness)
+                    if not direction:
+                        # Just draw whatever we can get if was not on the border
+                        poly = pya.DPolygon(points_2d)
+                    else:
+                        # Some points are duplicates when projected to 2D. Ensure you get two different points
+                        points_2d = list(set(points_2d))
+                        p1 = points_2d[0]
+                        p2 = points_2d[1]
+                        # Build a thick polygon from shifted points
+                        poly = pya.DPolygon(
+                            [
+                                p1,
+                                p2,
+                                p2 + direction,
+                                p1 + direction,
+                            ]
+                        )
+                        visualise_point = port.signal_location
+            else:
+                # Internal ports → use polygon directly
+                poly = pya.DPolygon(points_2d)
+
+            region = pya.Region(poly.to_itype(dbu))
+            self.visualise_region(region, label, "simulation_ports")
```

Comment:
```suggestion
            self.visualise_region(region, label, "simulation_ports", visualise_point)
```

You forgot to add `visualise_point` argument. Also handle `visualise_point` definition if the port is not an edgeport.

Please fix the indentation now, otherwise the code cannot be run. Use `black -l 120 -t py310 .` to fix formatting in the file. Then we can merge this PR